[fix 5496][master] fix npe

caoping6 opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->


## Purpose of the pull request

<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log
fix npe

this fix #5496

Commit:
修改注释

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
[fix 5496][master] fix npe

CalvinKirs commented:
Can you describe in detail what problem this PR solves?
BTW, please use English.

brave-lee commented on dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java:
```diff
@@ -323,6 +324,8 @@ private void executeComplementProcess() throws Exception {
             }
             // flow end
             // execute next process instance complement data
+            ProcessDefinition processDefinition = processService.findProcessDefineById(processDefinitionId);
```
Thanks for your contribution. Why is it need set processDefinition here? Can you describe the problems you encounter in detail

davidzollo replied:
> Thanks for your contribution. Why is it need set processDefinition here? Can you describe the problems you encounter in detail

I also have the same question, ^-^

echohlne replied:
@dailidong @JinyLeeChina  @caoping6 
It seems a same problem described in #5909
But it‘s better not to  modify at here.
I personally think that the problem comes in  [MasterExecThread.java#L977](https://github.com/apache/dolphinscheduler/blob/1887bde1ebf8249de153890e78d449582e3eaf4a/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java#L977)
aftert this:
`processInstance = processService.findProcessInstanceById(processInstance.getId());`
the `processDefinition` variable in `ProcessInstance` woule be null.
```java
// ProcessInstance.java
    /**
     * process definition structure
     */
    @TableField(exist = false)
    private ProcessDefinition processDefinition;
```

brave-lee commented on dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java:
```diff
@@ -930,7 +933,7 @@ private void runProcess() {
                 processAlertManager.sendProcessTimeoutAlert(processInstance,
                         processService.findProcessDefinition(processInstance.getProcessDefinitionCode(),
-                        processInstance.getProcessDefinitionVersion()));
+                                processInstance.getProcessDefinitionVersion()));
```
Please use unified codestyle

CalvinKirs commented:
Hi, how is it getting?

davidzollo committed:
Update MasterExecThread.java

code style config please refer the doc: https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html

davidzollo commented:
hi caoping6,
you have already done a good job when you submitted your first PR, it's the fist step for you guys to join the Contributor Community. 
I think you can contact me (wechat: taskflow or mail: lidongdai@apache.org) when you need any help

caishunfeng commented:
The latest code had been changed, so this pr will be closed.
